python/ql/src/semmle/python/regex.qll

```diff
@@ -124,16 +124,48 @@ abstract class RegexString extends Expr {
         )
     }
 
+    /** Escaped characters without any special handling (yet) */
+    private predicate singleEscape(int i) {
```

Comment:
At present, this predicate doesn't really do what it says on the tin. It holds for any index in the string that does not contain `x`, `u`, `U`, or `N`, _regardless_ of whether it's preceded by an escape character. 

Even though it's (in principle) a duplication of effort, I think it would be better to either reference `this.escapingChar` so that the documentation is correct, or simply inline this predicate where it's being used.

Reply:
With the rewrite using set literals, I could just inline it :)